internal/resource/config/oauth/idpadaptermappings/oauth_idp_adapter_mapping_resource.go

```diff
@@ -0,0 +1,28 @@
+package oauthidpadaptermappings
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (r *oauthIdpAdapterMappingResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	var plan *oauthIdpAdapterMappingResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if plan == nil {
+		return
+	}
+	// The idp adapter object will be configured based on the provided mapping_id
```

Comment:
This, strangely, is the opposite of how the authentication policy contract mapping version works. In that resource, the mapping_id is computed based on what is set in the idp_adapter_ref. This resource works exactly the opposite.

Maybe it makes sense for the provider to force one way or the other here?